### ISSUE TITLE
Short-term archiving not working if DOUT_S_SAVE_INTERIM_RESTART_FILES is false

### ISSUE THREAD

**Issue by billsacks** [ESMCI/cime#1572]
I just tried creating an A compset case from cime master, using create_newcase (not create_test). The short-term archiver failed with this error:

```
[Errno 2] No such file or directory: '/glade/scratch/sacks/archive/check_int_rest_0519a/rest/0001-01-06-00000/rpointer.ice'
```

I think the problem is this code in case_st_archive.py:

```python
    if not datename_is_last or case.get_value('DOUT_S_SAVE_INTERIM_RESTART_FILES') \
       or case.get_value("TEST"):
        if not os.path.exists(archive_restdir):
            os.makedirs(archive_restdir)
```

It seems to not create the necessary directory tree in a regular case with DOUT_S_SAVE_INTERIM_RESTART_FILES=FALSE.

Why does this have `or case.get_value("TEST")`? That is presumably why we haven't picked up any problems in our testing.

**Comment by rljacob** [comment]
CC @mfdeakin-sandia

**Comment by rljacob** [comment]
Isn't DOUT_S_SAVE_INTERIM_RESTART_FILES  set to FALSE by default?  The title would imply short term archiving stopped working at all and we know that isn't true.

**Comment by jedwards4b** [comment]
I think it was not working if REST_N < STOP_N and DOUT_S_SAVE_INTERIM_RESTART_FILES was FALSE

**Comment by rljacob** [comment]
I tried that in ACME recently and it worked fine.  With a fully coupled case.

**Comment by rljacob** [comment]
That's how I noticed the problem reported in #1449

**Comment by billsacks** [comment]
@jedwards4b - In my test case, I am pretty sure I had REST_N = STOP_N = 5 (days). So I'm pretty sure this bug applies to the default setup.

@rljacob - it was the fix for #1449 (i.e., PR #1511 ) that introduced this bug. So I suspect short-term archiving would be broken in ACME if you tried it with the current cime master.
